consensus/core/src/core.rs

```diff
@@ -614,10 +614,21 @@ impl Core {
             self.block_manager
                 .try_unsuspend_blocks_for_latest_gc_round();
 
-            committed_subdags.extend(subdags);
+            committed_sub_dags.extend(subdags);
         }
 
-        Ok(committed_subdags)
+        // Notify about our own committed blocks
```

Comment:
Btw can you expand a bit on how this notification is used? Since Sui already receives consensus commits that includes both blocks and transactions, why is notification here about committed blocks and transactions needed?

Reply:
I think it is ok to have this logic even if it is only used for metrics and tracking.

Reply:
Yeah so practically there were two options:
1. use the commit output to track what has been committed. However GC'ed blocks don't appear there. We could have some logic similar to this where we also calculate the gc round on the sui side etc, but I've avoided leaking such logic outside of consensus . Eventually would have to go with a solution like this https://github.com/MystenLabs/sui/pull/20011
2. Do it as this current PR is doing it where we might oversignal a bit similar info, but the whole logic is kept inside consensus and makes tracking on the Sui side easier. Also have stronger concurrency guarantees (ex we subscribe to status updates before even the block gets propagated). So I went for it as I found it more straightforward for now. We can refactor later if needed.